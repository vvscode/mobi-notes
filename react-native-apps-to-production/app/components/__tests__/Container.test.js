import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import { Component, styles, Container } from '../Container';

it('renders successfully', () => {
  const rendered = renderer.create(<Container />).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
  expect(typeof styles).toBe('object');
});

it('renders children props', () => {
  const rendered = renderer
    .create(
      <Container>
        <View />
      </Container>,
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

it('uses the specified backgroundColor, if provided', () => {
  const rendered = renderer.create(<Container backgroundColor="red" />).toJSON();
  expect(rendered).toMatchSnapshot();
});