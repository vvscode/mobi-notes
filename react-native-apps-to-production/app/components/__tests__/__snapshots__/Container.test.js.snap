// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children props 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View />
</View>
`;

exports[`uses the specified backgroundColor, if provided 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
